src/components/articles/ArticlePage.jsx

```diff
@@ -1,11 +1,48 @@
 import React from 'react'
+import { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom'
+import { getArticleById } from '../../api/api'
 
 const ArticlePage = () => {
+  const { article_id } = useParams();
+  const [ article, setArticle] = useState(null)
+  const [isLoading, setIsLoading] = useState(true)
+
+  useEffect(() => {
+    const fetchArticle = async () => {
+      try {
+        setIsLoading(true);
+        const articleData = await getArticleById(article_id);
+        setArticle(articleData.articles[0]);
+      } catch(err) {
+        console.error('Error fetching article', err)
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    fetchArticle();
```

Comment:
You've defined a function in this useEffect called `fetchArticle` and then you are immediately invoking the function on line 23, why not write the code inline inside the useEffect without a function wrapper ?

Reply:
it takes no arguments and returns nothing, just seems like extra steps for running the code